firmware/library/L1_Drivers/esp8266.hpp

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <array>
+#include <cstdint>
+
+#include "config.hpp"
+
+class Esp8266WifiInterface
+{
+ public:
+    std::array<char, config::kEsp8266BufferSize> default_buffer;
+    enum class ConnectionType : uint8_t
+    {
+        kTcp = 0,
+        kUdp = 1
+    };
+    virtual bool InitializeClientMode()                          = 0;
+    virtual bool TestModule()                                    = 0;
+    virtual void ResetModule()                                   = 0;
+    virtual void DeepSleep(uint32_t time)                        = 0;
+    virtual void GetModuleVersion()                              = 0;
+    virtual void GetNetworkConnection()                          = 0;
+    virtual bool ConnectToAccessPoint(
+        const char ssid[], const char password[])                = 0;
+    virtual void DisconnectFromAccessPoint()                     = 0;
+    virtual bool IsConnectedToAccessPoint()                      = 0;
+    virtual bool ConnectToRemoteIp(
+        const char remote_ip[], const char port[],
+        ConnectionType type = ConnectionType::kTcp)              = 0;
+    virtual void DisconnectFromRemoteIp()                        = 0;
+    virtual bool IsConnectedToRemoteIp()                         = 0;
+    virtual bool SendGetRequest(
+        const char url[], char buffer[] = default_buffer.data()) = 0;
+    virtual bool SendPostRequest(
```

Comment:
This and the above class should take a `size_t length` of the buffer. This needs to be known.